File: DSA/Shubham_590016506/Day_11_Question_3.c
#include <stdio.h>
#include <string.h>

int countSubstrings(char* s) {
    int n = strlen(s), count = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j <= 1; j++) {
            int l = i, r = i + j;
            while (l >= 0 && r < n && s[l] == s[r]) {
                count++;
                l--; r++;
            }
        }
    }
    return count;
}

int main() {
    char s[] = "aaa";
    printf("%d\n", countSubstrings(s));
}

File: DSA/LakshaKundra_590013237/Day_11_Question_3.py
def is_palindrome(s):
    return s == s[::-1]

def count_palindromic_substrings(s):
    count = 0
    n = len(s)

    for i in range(n):
        for j in range(i + 1, n + 1):
            if is_palindrome(s[i:j]):
                count += 1

    return count

print(count_palindromic_substrings("abc"))
print(count_palindromic_substrings("aaa"))
print(count_palindromic_substrings("aba"))
